internal/resource/config/oauth/oauth_auth_server_settings_data_source.go

```diff
@@ -0,0 +1,620 @@
+	state.AllowUnidentifiedClientROCreds = types.BoolPointerValue(r.AllowUnidentifiedClientROCreds)
+	state.AllowUnidentifiedClientExtensionGrants = types.BoolPointerValue(r.AllowUnidentifiedClientExtensionGrants)
+	state.AdminWebServicePcvRef, respDiags = resourcelink.ToDataSourceState(ctx, r.AdminWebServicePcvRef)
+	diags.Append(respDiags...)
+	state.AtmIdForOAuthGrantManagement = types.StringPointerValue(r.AtmIdForOAuthGrantManagement)
+	state.ScopeForOAuthGrantManagement = types.StringPointerValue(r.ScopeForOAuthGrantManagement)
+	state.AllowedOrigins = internaltypes.GetStringSet(r.AllowedOrigins)
+	state.UserAuthorizationUrl = types.StringPointerValue(r.UserAuthorizationUrl)
+	state.RegisteredAuthorizationPath = types.StringValue(r.RegisteredAuthorizationPath)
+	state.PendingAuthorizationTimeout = types.Int64Value(r.PendingAuthorizationTimeout)
+	state.DevicePollingInterval = types.Int64Value(r.DevicePollingInterval)
+	state.ActivationCodeCheckMode = types.StringPointerValue(r.ActivationCodeCheckMode)
+	state.BypassActivationCodeConfirmation = types.BoolValue(r.BypassActivationCodeConfirmation)
+	state.UserAuthorizationConsentPageSetting = types.StringPointerValue(r.UserAuthorizationConsentPageSetting)
+	state.UserAuthorizationConsentAdapter = types.StringPointerValue(r.UserAuthorizationConsentAdapter)
+	state.ApprovedScopesAttribute = types.StringPointerValue(r.ApprovedScopesAttribute)
+	state.ApprovedAuthorizationDetailAttribute = types.StringPointerValue(r.ApprovedAuthorizationDetailAttribute)
+	state.ParReferenceTimeout = types.Int64PointerValue(r.ParReferenceTimeout)
+	state.ParReferenceLength = types.Int64PointerValue(r.ParReferenceLength)
+	state.ParStatus = types.StringPointerValue(r.ParStatus)
+	state.ClientSecretRetentionPeriod = types.Int64PointerValue(r.ClientSecretRetentionPeriod)
+	state.JwtSecuredAuthorizationResponseModeLifetime = types.Int64PointerValue(r.JwtSecuredAuthorizationResponseModeLifetime)
+	return diags
+}
+
+// Read resource information
+func (r *oauthAuthServerSettingsDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state oauthAuthServerSettingsDataSourceModel
+
+	diags := req.Config.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	apiReadOauthAuthServerSettings, httpResp, err := r.apiClient.OauthAuthServerSettingsAPI.GetAuthorizationServerSettings(config.ProviderBasicAuthContext(ctx, r.providerConfig)).Execute()
+
+	if err != nil {
+		config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while getting the OAuth Auth Server Settings", err, httpResp)
+		return
+	}
+
+	// Log response JSON
+	_, responseErr := apiReadOauthAuthServerSettings.MarshalJSON()
+	if responseErr != nil {
+		diags.AddError("There was an issue retrieving the response of OAuth Auth Server Settings: %s", responseErr.Error())
+	}
+
+	// Read the response into the state
+	id := id.GenerateUUIDStringPointer()
```

Comment:
This is going to cause the id to change on every read, because we are generating a random UUID as the existing id passed in to the method below.